### PR TITLE
REG-6 Adds student, course, and student-course schemae

### PR DESCRIPTION
**Pull request by logandavis**
Jira: https://olinjs.atlassian.net/browse/REG-6
Jira: https://olinjs.atlassian.net/browse/REG-7

This is the first pass for the student, course, and student-course
instance schemae. They may need to be updated later in order to
make requirements really work, or to add sanitation beyond just typing.

### CODE REVIEW

**Comment by logandavis** [comment]
JIRA: https://olinjs.atlassian.net/browse/REG-6
https://olinjs.atlassian.net/browse/REG-7

**Review by Halliax — COMMENTED** [review]
Logan the fact that studentCourseSchema doesn't follow the naming convention bothers me. Otherwise this seems fine (as students, should we explicitly approve these? Or is a comment review like this fine)

File: models/studentCourseSchema.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+// Create student-course instance Schema
+var studentCourseSchema = mongoose.Schema({
+  studentId: Schema.Types.ObjectId,
+  courseId: Schema.Types.ObjectId,
+  semester: String,
+  grade: String
+});
+
+module.exports = mongoose.model("StudentCourse", studentCourseSchema);
```
Comment (billmwong) [inline models/studentCourseSchema.js]: newline at end of file

File: models/courseModel.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+var mongoose = require('mongoose');
+
+// Create course Schema
```
Comment (billmwong) [inline models/courseModel.js]: I'd remove these comments, it doesn't tell the reader any more info than the code does

File: models/courseModel.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+var courseSchema = mongoose.Schema({
+  registrarId: String,
+  title: String,
+  credits: Number,
+  ENGRCredits: Number,
+  AHSCredits: Number,
+  MTHCredits: Number,
+  SCICredits: Number,
+  ECredits: Number,
+  NDGRCredits: Number,
```
Comment (billmwong) [inline models/courseModel.js:12]: Remind me what `NDGR` stands for?

File: models/studentCourseSchema.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+var mongoose = require('mongoose');
+var Schema = mongoose.Schema;
```
Comment (billmwong) [inline models/studentCourseSchema.js]: Don't need to set this variable if you specify `mongoose.Schema` below

**Review by billmwong — CHANGES_REQUESTED** [review]
In addition to the in-line comments, update the syntax to be es6 please!  Look at my wiki entry [here](https://github.com/olin/register/wiki/ES6-Basics) for the basics on what you need to know for es6.

Also remember to name your PRs with your Jira ticket number in it.

File: models/courseModel.js (hunk 3 above)
Comment (logandavis) [reply models/courseModel.js:12]: Non-degree.

**Comment by logandavis** [comment]
@Halliax which naming conventions does studentCourseSchema break? I'm looking at this: https://google.github.io/styleguide/jsoncstyleguide.xml

File: models/studentCourseSchema.js (hunk 1 above)
Comment (logandavis) [reply models/studentCourseSchema.js]: Github didn't catch that my new commit obsoleted this comment, but it did. Done.

**Review by billmwong — CHANGES_REQUESTED** [review]
Sorry to be nitpicky, but rebase from master and run `npm start` to see the backend linter on your files here, and fix the errors.  Mostly using single-quote strings and comma dangles.  Also rebasing will fix the merge conflicts.

**Comment by Halliax** [comment]
@logandavis I just meant the file name, with your other filenames. The other schema files are "model"

**Review by billmwong — APPROVED** [review]

**Pull request merged**